infra/templates/az-isolated-service-single-region/app.tf

```diff
@@ -0,0 +1,35 @@
+// This file contains all of the resources that exist within the app dev subscription. Design documentation
+// with more information on exactly what resources live here can be found at ./docs/design.md
+
+// Note: unfortunately the alias cannot be configured by passing a variable through
+// the module initialization!
+provider "azurerm" {
+  alias           = "app_dev"
+  subscription_id = var.app_dev_subscription_id
```

Comment:
Where does var.app_dev_subscription_id derive it's value? I'm not seeing any calls to the env anywhere.

Reply:
This is defined in `infra/templates/az-isolated-service-single-region/variables.tf`. The value will come by supplying a `-var` param when running `terraform plan|apply` or by declaring `TF_VAR_app_dev_subscription_id` as an environment variable, which is done in the build pipeline. The variable for the build pipeline is configured in AZDO as an environment variable.

I also have decided to use the env-var approach since its easier on my local. Its not in the `.env.template`, though, since it is not actually required as an environment variable.

Reply:
Is the intention command line only?

Reply:
Not necessarily. There are a number of options that you may want to use depending on your deployment strategy. See below. I think that the implementation of this was not covered in any design thus far

https://www.terraform.io/docs/configuration/variables.html#assigning-values-to-root-module-variables

Reply:
Have we put this in our docs for devs?

Reply:
No, there is no readme for this template yet but I think it would be a good idea to add it to the README when we start documentation.

Perhaps the documentation should already have been started but I was planning on writing it up after things normalize with the inputs/outputs and whatnot

Reply:
I'll do this as part of https://github.com/microsoft/cobalt/issues/204